kernel/common/numeric.rb

```diff
@@ -31,23 +31,27 @@ def <=>(other)
     return nil
   end
 
-  def step(limit, step=1)
+  def step(limit=nil, step=nil, to: nil, by: nil)
     unless block_given?
-      return to_enum(:step, limit, step) do
-        Rubinius::Mirror::Numeric.reflect(self).step_size(limit, step)
+      return to_enum(:step, limit, step, to: to, by: by) do
+        Rubinius::Mirror::Numeric.reflect(self).step_size(limit, step, to, by)
       end
     end
 
-    raise ArgumentError, "step cannot be 0" if step == 0
```

Comment:
This error is still raised in case of the following:

    10.step(100, 0).to_a => ArgumentError: step can't be 0